src/plugins/data/common/data_frames/utils.ts

```diff
@@ -472,3 +497,70 @@ export const dataFrameToSpec = (dataFrame: IDataFrame, id?: string): IndexPatter
     fields: fields.reduce(flattenFields, {} as IndexPatternFieldMap),
   };
 };
+
+export class Polling<T, P = void> {
```

Comment:
nit: in my opinion it's a little bit weird to call it Polling if it just takes takes params and has functions.

unless we extend it it from DataFrame and then we could call like dataFramePolling.fetch